docs/specs/ML.Net Sweepable API.md

```diff
@@ -0,0 +1,162 @@
+                .Append(context.Transforms.Concatenate(@"TextFeature", @"Name", "Ticket", "Cabin"))
+                .Append(context.Sweepable().CreateSweepableEstimator(
+                    (mlContext, option) =>
+                    {
+                        var textOption = new TextFeaturizingEstimator.Options
+                        {
+                            CaseMode = option.CaseMode,
+                            KeepDiacritics = option.KeepDiacritics,
+                            KeepNumbers = option.KeepNumbers,
+                            KeepPunctuations = option.KeepPunctuations,
+                            CharFeatureExtractor = new WordBagEstimator.Options()
+                            {
+                                NgramLength = option.WordBagEstimatorOption.NgramLength,
+                                UseAllLengths = option.WordBagEstimatorOption.UseAllLengths,
+                                Weighting = option.WordBagEstimatorOption.WeightingCriteria,
+                            },
+                        };
+
+                        return mlContext.Transforms.Text.FeaturizeText("TextFeature", textOption);
+                    },
+                    textFeaturizeSS))
+                .Append(context.Transforms.Concatenate(@"Features", new[] { @"Sex", @"Embarked", @"Pclass", @"Age", @"SibSp", @"Parch", @"Fare", "TextFeature" }))
+                .Append(context.Transforms.Conversion.ConvertType("Survived", "Survived", Data.DataKind.Boolean))
+                .Append(context.Sweepable().CreateSweepableEstimator(
+                    (mlContext, option) => mlContext.BinaryClassification.Trainers.FastForest(labelColumnName: "Survived", featureColumnName: "Features", numberOfLeaves: option.NumberOfLeavenumberOfTrees: option.NumberOfTrees),
+                    fastTreeSS))
+                .Append(context.BinaryClassification.Calibrators.Naive(labelColumnName: @"Survived", scoreColumnName: @"Score"));
+```
+
+After sweepable pipeline is created, one can call `BuildTrainingPipeline` to convert it to a ml.net pipeline.
+
+## tuner
+
+tuner takes in a search space and performs hpo algos. There're a few default tuning alogs provided by Sweepable API (grid search/random search), and there'll be more smart hpo algos coming soon.
+
+The way to use a tuner is quite similar with the way to use an enumerator. The code below shows how tuner works with search space and sweepable estimator/pipeline
+
+```csharp
+var ss = pipeline.SearchSpace
+var tuner = new GridSearchTuner(ss);
+var df = DataFrame.LoadCsv(@"titanic.csv");
+var trainTestSplit = context.Data.TrainTestSplit(df, 0.1);
+var bestAccuracy = 0.0;
+var i = 0;
+foreach (var param in tuner.Propose())
+{
+    Console.WriteLine($"trial {i++}");
+
+    // convert sweepable pipeline to ml.net pipeline
+    var trainingPipeline = pipeline.BuildTrainingPipeline(context, param);

```

Comment:
Just curious, but would there ever be a need to save the current state of the sweeper so that it can be stopped and then resumed at a later date?

Reply:
If sweeper supports that then yes. In fact continue training has already been supported in model builder cfo tuner, it's just not exposed yet.

Reply:
Maybe for the future, but how about "import experiments"? That would help in case someone wants to distribute training to multiple nodes: a) save each experiment results to db  b) before starting next iteration, fetch new from db and import

Reply:
@torronen 
That's a wonderful suggestion, thanks!